modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectEntries/ScheduleContainer.spec.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import ScheduleContainer from '../../../object_entries/object_entry/ScheduleContainer';
+
+function renderScheduleContainer(
+	scheduledProperties = {
+		reviewDate: {
+			checked: false,
+			value: '',
+		},
+	},
+	portletNamespace = ''
+) {
+	return render(
+		<>
+			<ScheduleContainer
+				portletNamespace={portletNamespace}
+				scheduledProperties={scheduledProperties}
+			/>
+			<button data-testid="blur-target">Blur Target</button>
+		</>
+	);
+}
+
+describe('ScheduleContainer component', () => {
+	beforeAll(() => {
+		global.Liferay = {
+			...global.Liferay,
+			FeatureFlags: {
+				...global.Liferay?.FeatureFlags,
+				'LPD-17564': true,
+			},
+		};
+	});
+
+	it('reviewDate is disabled when checked is true', async () => {
+		const {container} = renderScheduleContainer({
+			reviewDate: {
+				checked: true,
+				value: '',
+			},
+		});
+
+		const checkbox = screen.getByRole('checkbox');
+		const reviewDateInput = container.querySelector(
+			'input[id$="reviewDate"]'
+		);
+
+		expect(reviewDateInput).toBeDisabled();
+		expect(checkbox).toBeChecked();
+	});
+
+	it('shows required error on blur when no value is provided', async () => {
+		const {container} = renderScheduleContainer({
+			reviewDate: {
+				checked: false,
+				value: '',
+			},
+		});
+
+		const reviewDateInput = container.querySelector(
+			'input[id$="reviewDate"]'
+		) as HTMLElement;
+
+		await userEvent.click(reviewDateInput);
+
+		await userEvent.click(screen.getByTestId('blur-target'));
+
+		const errorMessage = screen.getByText('this-field-is-required');
+
+		expect(errorMessage).toBeInTheDocument();
+	});
+
+	it('shows required error on change when no value is provided', async () => {
+		const {container} = renderScheduleContainer({
+			reviewDate: {
+				checked: false,
+				value: '05/13/2025 02:38 PM',
+			},
+		});
+
+		const reviewDateInput = container.querySelector(
+			'input[id$="reviewDate"]'
+		) as HTMLElement;
+
+		await userEvent.click(reviewDateInput);
+
+		await userEvent.type(reviewDateInput, '{selectall}{backspace}');
```

Comment:
Is this the expected behavior? If the user is still focused on the field and just cleared the input, then I think the error shouldn't be displayed because he may still input some value. I would display error only on blur.

Reply:
Yes, the objects datePicker is slightly different  from the forms one 😅 
When we try to delete the content, it deletes all of it instead of just a character, that's why we're validating it that way. I wanted to find the cause for this behavior and fix it, but I didn't have the time to do it within the scope of this story.